File: app/src/main/java/projects/feng/gary/matrixtools/MainActivity.java
package projects.feng.gary.matrixtools;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.rrefButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                TextView text = findViewById(R.id.text);

                Fraction[] test = {new Fraction(5), new Fraction(3), new Fraction(2),
                                   new Fraction("3/2"), new Fraction(0), new Fraction("1/2"),
                                   new Fraction(3), new Fraction(0), new Fraction(1),
                                   new Fraction(6), new Fraction(0), new Fraction(2)};
                Matrix matrix = new Matrix(test, 4, 3);
                Fraction[] rref = matrix.RREF();

                for (int i = 0; i < 4; ++i) {
                    for (int j = 0; j < 3; ++j) {
                        text.append(rref[matrix.getIndex(i, j)] + "\t\t");
                    }
                    text.append("\n");
                }
            }
        });
    }


}

File: app/src/main/java/projects/feng/gary/matrixtools/MainActivity.kt
package projects.feng.gary.matrixtools

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setSupportActionBar(homeBar as Toolbar)
        supportActionBar?.title = getString(R.string.home_title)

        initSpinners()

        val startButton = startToolButton
        startButton.setOnClickListener({ _ ->
            val numRows = rowSpinner.selectedItem as Int
            val numCols = colSpinner.selectedItem as Int

            val intent = Intent(this, CalculatorActivity::class.java).apply {
                putExtra(EXTRA_NUM_ROWS, numRows)
                putExtra(EXTRA_NUM_COLS, numCols)
            }

            startActivity(intent)
        })
    }

    private fun initSpinners() {
        val dimensionAdapter = ArrayAdapter<Int>(this, R.layout.spinner_holder)
        dimensionAdapter.setDropDownViewResource(R.layout.spinner_row)
        rowSpinner.adapter = dimensionAdapter
        val possibleRows = listOf(1, 2, 3, 4, 5, 6)
        dimensionAdapter.addAll(possibleRows)

        colSpinner.adapter = dimensionAdapter
    }

}
